src/seml/project.py

```diff
@@ -102,30 +115,63 @@ def ignore_file(file_path: str):
     logging.info('Project initialized successfully')
 
 
-def get_available_templates():
+def checkout_template_repo(
+    git_remote: Optional[str] = None, git_commit: Optional[str] = None
+) -> 'Repo':
+    from git import Repo
+
+    if git_remote is None:
+        git_remote = SETTINGS.TEMPLATE_REMOTE
+
+    tmp_dir = Path(SETTINGS.TMP_DIRECTORY) / str(uuid.uuid4())
```

Comment:
Use `tempfile.TemporaryDirectory(delete=False)` and delete it after usage?
Or write this as a context manager that uses `tempfile.TemporaryDirectory` and upon `__exit__` exits the `tempfile` context (most clean option imo)